Lib/test/test_typing.py

```diff
@@ -2654,6 +2654,42 @@ class D(PNonCall): ...
         with self.assertRaises(TypeError):
             issubclass(D, PNonCall)
 
+    def test_no_weird_caching_with_issubclass_after_isinstance(self):
+        @runtime_checkable
+        class Spam(Protocol[T]):
```

Comment:
The generics aren't essential to this issue, right? Would be good to add a non-generic test for simplicity.

Reply:
True, I just needed the generics in the other test so that I could test PEP-695 syntax!